spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/cassandra/CassandraReactiveHealthIndicator.java

```diff
@@ -15,43 +15,47 @@
  */
 package org.springframework.boot.actuate.cassandra;
 
+import java.util.Objects;
+
 import com.datastax.oss.driver.api.core.ConsistencyLevel;
+import com.datastax.oss.driver.api.core.CqlSession;
 import com.datastax.oss.driver.api.core.cql.SimpleStatement;
 import reactor.core.publisher.Mono;
 
 import org.springframework.boot.actuate.health.AbstractReactiveHealthIndicator;
 import org.springframework.boot.actuate.health.Health;
 import org.springframework.boot.actuate.health.ReactiveHealthIndicator;
-import org.springframework.data.cassandra.core.ReactiveCassandraOperations;
 import org.springframework.util.Assert;
 
 /**
  * A {@link ReactiveHealthIndicator} for Cassandra.
  *
  * @author Artsiom Yudovin
+ * @author Alexandre Dutra
  * @since 2.1.0
  */
 public class CassandraReactiveHealthIndicator extends AbstractReactiveHealthIndicator {
 
 	private static final SimpleStatement SELECT = SimpleStatement
 			.newInstance("SELECT release_version FROM system.local").setConsistencyLevel(ConsistencyLevel.LOCAL_ONE);
 
-	private final ReactiveCassandraOperations reactiveCassandraOperations;
+	private final CqlSession session;
 
 	/**
 	 * Create a new {@link CassandraHealthIndicator} instance.
-	 * @param reactiveCassandraOperations the Cassandra operations
+	 * @param session the {@link CqlSession}.
 	 */
-	public CassandraReactiveHealthIndicator(ReactiveCassandraOperations reactiveCassandraOperations) {
+	public CassandraReactiveHealthIndicator(CqlSession session) {
 		super("Cassandra health check failed");
-		Assert.notNull(reactiveCassandraOperations, "ReactiveCassandraOperations must not be null");
-		this.reactiveCassandraOperations = reactiveCassandraOperations;
+		Assert.notNull(session, "session must not be null");
+		this.session = session;
 	}
 
 	@Override
 	protected Mono<Health> doHealthCheck(Health.Builder builder) {
-		return this.reactiveCassandraOperations.getReactiveCqlOperations().queryForObject(SELECT, String.class)
-				.map((version) -> builder.up().withDetail("version", version).build()).single();
+		return Mono.from(this.session.executeReactive(SELECT))
+				.map((row) -> Objects.requireNonNull(row.getString(0), "release_version should never be null"))
```

Comment:
Same here, `Objects.requireNonNull` is just making my IDE happy.

Reply:
IntelliJ IDEA? I think that's a false positive. 

```java
return Mono.from(this.session.executeReactive(SELECT))
		.map((row) -> builder.up().withDetail("version", row.getString(0)).build());
```

this is more concise and doesn't yield any warning for me.

Reply:
Actually `GettableByIndex#getString(int)` is marked `@Nullable` in the driver API, so it may trigger a warning depending on how you configure things. But again I don't mind really, I'll do as you suggest.

Reply:
I think that's going a tad too far. The `row` will never be `null` given the contract of `Publisher` and `onNext`.